tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -1837,3 +1838,56 @@ def test_nested_build(self, use_container, cached, parallel):
                     ("LocalNestedStack/Function2", {"pi": "3.14"}),
                 ],
             )
+
+
+@skipIf(
+    ((IS_WINDOWS and RUNNING_ON_CI) and not CI_OVERRIDE),
+    "Skip build tests on windows when running in CI unless overridden",
+)
+class TestBuildWithCustomBuildImage(BuildIntegBase):
+    template = "provided_image_function.yaml"
+
+    @parameterized.expand(
+        [
+            ("use_container", None),
+            ("use_container", "amazon/aws-sam-cli-build-image-nodejs10.x"),
+        ]
+    )
+    @pytest.mark.flaky(reruns=3)
+    def test_custom_build_image_succeeds(self, use_container, build_image):
+        if use_container and SKIP_DOCKER_TESTS:
+            self.skipTest(SKIP_DOCKER_MESSAGE)
+
+        cmdlist = self.get_command_list(use_container=use_container, build_image=build_image)
+
+        LOG.info("Running Command: {}".format(cmdlist))
+        process = Popen(cmdlist, cwd=self.working_dir, stdout=PIPE, stderr=PIPE)
+        try:
+            stdout, stderr = process.communicate(timeout=TIMEOUT)
+            LOG.info(f"Stdout: {stdout.decode('utf-8')}")
```

Comment:
Do we need this info always displayed in our test output?

Same goes for other `LOG.` statements, mainly those who will output logs of data (stderr and stdout).

Reply:
We do this same logging behaviour for all of our integration tests, when we run the command @jfuss

Reply:
In other build tests we do not log stdout and stderr. This comment is about removing the large outputs in our tests logs. Leaving what commands and other small information is fine, logging 30+ lines seems like overkill. If you think this info is important, then we should only output it on test failure instead of every single run.

Reply:
I guess we do but it is done through the `run_command`: https://github.com/aws/aws-sam-cli/blob/4543732bf3c0da3b57fe1e5aa43ce3f41d2bd0ba/tests/testing_utils.py#L25 Why are we not using this here?

Reply:
because we need to do extra work that run_command does not provide, which is checking the content in the stderr

Reply:
We do support checking `stderr` using `run_command`

https://github.com/aws/aws-sam-cli/blob/4543732bf3c0da3b57fe1e5aa43ce3f41d2bd0ba/tests/integration/buildcmd/test_build_cmd.py#L1548-L1551

Reply:
I can try something like this, trying to use the nametuple returned

Reply:
Before I had several differences between this function and run command, but seems like this run_command will return what I need

Reply:
Updated to use run_command()